src/CachePool.php

```diff
@@ -0,0 +1,388 @@
+<?php
+declare(strict_types=1);
+
+namespace WpOop\TransientCache;
+
+use DateInterval;
+use DateTimeImmutable;
+use Exception;
+use Psr\SimpleCache\CacheInterface;
+use RangeException;
+use wpdb;
+use WpOop\TransientCache\Exception\CacheException;
+use WpOop\TransientCache\Exception\InvalidArgumentException;
+
+/**
+ * {@inheritDoc}
+ *
+ * Uses WordPress transients as storage medium.
+ */
+class CachePool implements CacheInterface
+{
+    public const RESERVED_KEY_SYMBOLS = '{}()/\@:';
+    public const NAMESPACE_SEPARATOR = '/';
+
+    protected const TABLE_NAME_OPTIONS = 'options';
+    protected const FIELD_NAME_OPTION_NAME = 'option_name';
+    protected const OPTION_NAME_PREFIX_TRANSIENT = '_transient_';
+    protected const OPTION_NAME_PREFIX_TIMEOUT = 'timeout_';
+
+    /**
+     * @var wpdb
+     */
+    protected $wpdb;
+    /**
+     * @var string
+     */
+    protected $poolName;
+    /**
+     * @var mixed
+     */
+    protected $defaultValue;
+
+    /**
+     * @param wpdb   $wpdb         The WP database object.
+     * @param string $poolName     The name of this cache pool. Must be unique to this instance.
+     * @param mixed $defaultValue  A random value. The more chaotic - the better.
```

Comment:
Consider mentioning that this is used for false-negative detection.

Reply:
Done.